File: frontend/src/app/shared/class/address.ts
export class Address {
    address_id: string;
    line_1: string;
    line_2: string;
    country: string;
    state: string;
    city: string;
    pin_code: number;

    constructor(address_id: string, line_1: string, line_2: string, country: string, state: string, city: string, pin_code: number){
        this.address_id = address_id;
        this.line_1 = line_1;
        this.line_2 = line_2;
        this.country = country;
        this.state = state;
        this.city = city;
        this.pin_code = pin_code;

    }

    set address_Id(address_id: string){
        this.address_id = address_id;
    }

    get address_Id(): string{
        return this.address_id;
    }

    set address_line1(line_1: string){
        this.line_1 = line_1;
    }

    get address_line1(): string{
        return this.line_1;
    }

    set address_line2(line_2: string){
        this.line_2 = line_2;
    }

    get address_line2(): string{
        return this.line_2;
    }

    set Country(country: string){
        this.country = country;
    }

    get Country(): string{
        return this.country;
    }

    set State(state: string){
        this.state = state;
    }

    get State(): string{
        return this.state;
    }

    set City(city: string){
        this.city = city;
    }

    get City(): string{
        return this.city;
    }

    set pincode(pin_code: number){
        this.pin_code = pin_code;
    }

    get pincode(): number{
        return this.pin_code;
    }


}

File: backend/src/main/java/com/vencent/hotelmanagement/entity/Address.java
package com.vencent.hotelmanagement.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.Table;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.OneToOne;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name="address")
public class Address {

	@Id
	@GeneratedValue
	@Column(name="address_id")
	private int addressId;
	
	@Column(name="line_1")
	private String line1;
	
	@Column(name="line_2")
	private String line2;
	
	private String city;
	
	@Column(name="pin_code")
	private int pinCode;
	
	private String country;
	
	private String state;
	
	@OneToOne(fetch=FetchType.EAGER,cascade=CascadeType.ALL)
	@JoinColumn(name = "type_id", nullable = false)
	private Type type;

	public int getAddressId() {
		return addressId;
	}

	public void setAddressId(int addressId) {
		this.addressId = addressId;
	}

	public String getLine1() {
		return line1;
	}

	public void setLine1(String line1) {
		this.line1 = line1;
	}

	public String getLine2() {
		return line2;
	}

	public void setLine2(String line2) {
		this.line2 = line2;
	}

	public String getCity() {
		return city;
	}

	public void setCity(String city) {
		this.city = city;
	}

	public int getPinCode() {
		return pinCode;
	}

	public void setPinCode(int pinCode) {
		this.pinCode = pinCode;
	}

	public String getCountry() {
		return country;
	}

	public void setCountry(String country) {
		this.country = country;
	}

	public String getState() {
		return state;
	}

	public void setState(String state) {
		this.state = state;
	}

	public Type getType() {
		return type;
	}

	public void setType(Type type) {
		this.type = type;
	}
	
	
	
}



